flaky Android build

The Android build will sometimes fail with the following output:

```
make[2]: *** No rule to make target `Release/obj.target', needed by `/bitrise/src/platform/android/MapboxGLAndroidSDK/src/main/obj.target'.  Stop.
make[2]: *** Waiting for unfinished jobs....
  CXX(target) Release/obj.target/example-custom-layer-lib/platform/android/src/example_custom_layer.o
../../platform/android/src/http_request_android.cpp:191:59: warning: unused parameter 'message' [-Wunused-parameter]
void HTTPAndroidRequest::onResponse(int code, std::string message, std::string etag, std::string modified, std::string cacheControl, std::string expires, std::string body) {
                                                          ^
1 warning generated.
make[2]: Leaving directory `/bitrise/src/build/android-arm-v7'
make[1]: Leaving directory `/bitrise/src'
make[1]: *** [Makefile/androidapp] Error 2
make: *** [android-lib] Error 2
==> Script finished with exit code: 2
ERRO[21:59:51] Step (Build Android SDK For arm) failed, error: (exit status 2) 
```

I don't think the "warning: unused parameter" is relevant (although we should fix that). I have no idea what "No rule to make target `Release/obj.target'" means, but it seems to be the cause. There's nothing before that.

--- a/platform/android/src/http_request_android.cpp
+++ b/platform/android/src/http_request_android.cpp
@@ -192,6 +192,9 @@ void HTTPAndroidRequest::onResponse(int code, std::string message, std::string e
     response = std::make_unique<Response>();
     using Error = Response::Error;
 
+    // the message param is unused, this generates a warning at build time
+    // this was breaking builds for `make android -j4`
+    (void)message;
     response->modified = Seconds(parse_date(modified.c_str()));
     response->etag = etag;
     response->expires = parseCacheControl(cacheControl.c_str());

